ui_framework/src/components/header/_header.scss

```diff
@@ -0,0 +1,158 @@
+// Header. Includes breadcrumbs of nav buttons.
+
+$kuiHeaderChildSize: $kuiSizeXXL + $kuiSizeL;
+
+@include component('kuiHeader') {
+  background: $kuiHeaderBackgroundColor;
+  border-bottom: $kuiBorderThin;
+
+  @include child('logo') {
+    position: relative;
+    height: $kuiHeaderChildSize;
+    padding: $kuiSize $kuiSizeXL;
+    display: inline-block;
+    vertical-align: middle;
+
+    &:after {
+      position: absolute;
+      content: "";
+      width: 1px;
+      top: $kuiSize;
+      bottom: 0;
+      right: 0;
+      background: $kuiBorderColor;
+    }
```

Comment:
Because the styles for the vertical separator is duplicated, I suggest we just create a wrapper element called `kuiHeaderItem` which has these styles. Then we'd use it as a container for the logo and the buttons:

```html
<div class="kuiHeaderItem">
  <a href="" class="kuiHeader__logo">
    <svg class="kuiIcon kuiHeader__logoIcon">
      <use href="#logo" />
    </svg>
  </a>
</div>
```

```html
<div class="kuiHeaderItem">
  <button class="kuiHeader__button">
    <svg class="kuiIcon kuiIcon--medium">
      <use href="#search" />
    </svg>
  </button>
</div>

<div class="kuiHeaderItem">
  <button class="kuiHeader__button">
    <svg class="kuiIcon kuiIcon--medium uVerticalAlignMiddle">
      <use href="#user" />
    </svg>
    <span class="kuiHeader__notification">3</span>
  </button>
</div>

<div class="kuiHeaderItem">
  <button class="kuiHeader__button">
    <svg class="kuiIcon kuiIcon--medium">
      <use href="#apps" />
    </svg>
  </button>
</div>
```

This leverages [composition](https://en.wikipedia.org/wiki/Composition_over_inheritance) which I think will make it easier to change and customize the header in the future, e.g. by adding new items or changing the appearance of all items.